### ISSUE TITLE
Github release should not create lightweight tags

### ISSUE THREAD

**Issue by ssbarnea** [isaacs/github#1916]
At this moment creating a release (tag) using releases page does only create a lightweight git tag, which does not contain essential information like time when it was created. This issue created bug like https://github.com/pypa/setuptools_scm/issues/521 as tools are not able to get information about the tags.

I do think that any tag created by github release should be annotated one: `git tag -a` instead of just `git tag`, even if the message would be empty or hardcoded, as this would provide the timestamps that are heavily needed.

Take a look at the difference between annotate tags and lightweight ones, the `creatordate` is not the tagging date, is the date of the commit! Only annotated ones have the `taggerdate` property.
```
$ git tag --format='%(creatordate)%09%(refname:strip=2)   %(taggerdate)'                                                                           [10:28:48]
Tue Feb 9 09:48:55 2021 +0000	0.1
Tue Feb 9 09:48:55 2021 +0000	0.1.1
Tue Feb 9 08:55:44 2021 +0000	0.1a0
Tue Feb 9 09:21:49 2021 +0000	0.1a1
Tue Feb 9 09:48:55 2021 +0000	0.1a2
Tue Feb 9 09:48:55 2021 +0000	0.1a3
Tue Feb 9 09:48:55 2021 +0000	0.1a4
Tue Feb 9 10:16:42 2021 +0000	0.1a5   Tue Feb 9 10:16:42 2021 +0000
Tue Feb 9 10:19:46 2021 +0000	0.1a6   Tue Feb 9 10:19:46 2021 +0000
Tue Feb 9 10:25:51 2021 +0000	0.2   Tue Feb 9 10:25:51 2021 +0000
Tue Feb 9 10:25:59 2021 +0000	0.2.1   Tue Feb 9 10:25:59 2021 +0000
Tue Feb 9 09:21:49 2021 +0000	0.3a0
```

**Comment by TPS** [comment]
@ssbarnea I hope you don't mind, but I've adjusted your issue title to include the asked-for solution (which I've a strong preference for — I'd hope such would be acted upon sooner) rather than just the problem.

**Comment by psigen** [comment]
I agree strongly with this, can we post it on the official feedback repo?
https://github.com/github/feedback/discussions/categories/general-feedback

**Comment by psigen** [comment]
I created one and referenced it here:
https://github.com/github/feedback/discussions/4924
